src/lib/classes/styling/GlobalCssSchemesLoader.ts

```diff
@@ -0,0 +1,206 @@
+import type ColorValue from "./ZodSchemas/AttributeSchemas/ColorAttribute";
+import type MediaScheme from "./ZodSchemas/MediaScheme";
+
+import MediaSchemes from "./ZodSchemas/MediaSchemes";
+import GlobalCssProperties from "../../GlobalCssProperties.json";
+
+import type { z } from "zod";
+
+/**
+ * Class for handling the loading of different properties based on active media features
+ */
+class GlobalCssSchemesLoader {
+	private _window: Window;
+	private _mediaSchemes: z.infer<typeof MediaSchemes>;
+	private _propertyNames: string[] = [];
+
+	// SUPPORTED MEDIA FEATURES
+	private _supportedMediaFeatures: string[] = [
+		"prefers-color-scheme: dark",
+		"prefers-color-scheme: light",
+		"prefers-reduced-motion",
+		"prefers-reduced-transparency",
+	];
+
+	constructor(window: Window) {
+		this._window = window;
```

Comment:
```suggestion
		if(!("window" in globalThis)) throw new Error("The CSS loader needs access to the window and DOM elements");
```

I don't see why you would need to import the window from a different context. The class should never be, and is never, instantiated without access to the global `window` property.

You can instead write this check to give a developer-friendly error message if they do instantiate it in the wrong place.